### PR TITLE
Increase probability ramble will continue in broken workspace

### PR DESCRIPTION
**Pull request by rfbgo**
This PR does 2 main things:

1. It adds a new error type to our existing early exit handling around broken workspace configs, to allow bad syntax not to block `ramble workspace edit`
2. It allows ramble to not hard exit (but still warn) on a larger set of commands

### CODE REVIEW

File: lib/ramble/ramble/main.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import traceback
 import warnings
 import jsonschema
+import ruamel
```
Comment (rfbgo) [inline lib/ramble/ramble/main.py:28]: It's possible there a nice way to alias out that error to not need a ruamel dep and to get it via the json stuff

Comment (douglasjacobsen) [reply lib/ramble/ramble/main.py:28]: Do you know where (in Ramble) it originates from?

Comment (rfbgo) [reply lib/ramble/ramble/main.py:28]: Yea it comes out of workspace calling `syaml.load_config`. Stack trace:

```
Traceback (most recent call last):
  File "/Users/robertbird/ramble/bin/ramble", line 70, in <module>
    sys.exit(ramble.main.main())
  File "/Users/robertbird/ramble/lib/ramble/ramble/main.py", line 880, in main
    return _main(argv)
  File "/Users/robertbird/ramble/lib/ramble/ramble/main.py", line 787, in _main
    ws = ramble.cmd.find_workspace(args)
  File "/Users/robertbird/ramble/lib/ramble/ramble/cmd/__init__.py", line 279, in find_workspace
    return ramble.workspace.Workspace(ws)
  File "/Users/robertbird/ramble/lib/ramble/ramble/workspace/workspace.py", line 519, in __init__
    self._read()
  File "/Users/robertbird/ramble/lib/ramble/ramble/workspace/workspace.py", line 558, in _read
    self._read_config(config_section, f)
  File "/Users/robertbird/ramble/lib/ramble/ramble/workspace/workspace.py", line 631, in _read_config
    self._read_yaml(config, f, raw_yaml)
  File "/Users/robertbird/ramble/lib/ramble/ramble/workspace/workspace.py", line 661, in _read_yaml
    config['raw_yaml'], config['yaml'] = _read_yaml(f,
  File "/Users/robertbird/ramble/lib/ramble/ramble/workspace/workspace.py", line 1416, in _read_yaml
    data = syaml.load_config(str_or_file)
  File "/Users/robertbird/ramble/lib/ramble/spack/util/spack_yaml.py", line 300, in load_config
    return yaml.load(*args, **kwargs)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/main.py", line 75, in load
    return loader.get_single_data()
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/constructor.py", line 65, in get_single_data
    node = self.get_single_node()
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 53, in get_single_node
    document = self.compose_document()
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 73, in compose_document
    node = self.compose_node(None, None)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 104, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 163, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 102, in compose_node
    node = self.compose_sequence_node(anchor)
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/composer.py", line 132, in compose_sequence_node
    while not self.check_event(SequenceEndEvent):
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/parser.py", line 116, in check_event
    self.current_event = self.state()
  File "/Users/robertbird/ramble/lib/ramble/external/ruamel/yaml/parser.py", line 553, in parse_flow_sequence_entry
    raise ParserError(
ruamel.yaml.parser.ParserError: while parsing a flow sequence
```

Comment (douglasjacobsen) [reply lib/ramble/ramble/main.py:28]: Could we catch the ruamel error closer to that load_config call, and then convert it into a generic config format error?

Comment (rfbgo) [reply lib/ramble/ramble/main.py:28]: We could, be actually prefer it being the "true" ruamel error, not some sneaky version of that

The info it gives you is also super useful and digestable, tells you the line etc. I don't want to change the error just allow it through when we do edit

**Pull request merged**